src/bio/bio_context.c

```diff
@@ -289,7 +289,7 @@ bio_blob_delete(uuid_t uuid, struct bio_xs_context *xs_ctxt)
 	/**
 	 * Query per-server metadata to get blobID for this pool:target
 	 */
-	rc = smd_pool_get_blob(uuid, xs_ctxt->bxc_tgt_id, &blob_id);
+	rc = smd_pool_get_blob(uuid, xs_ctxt->bxc_tgt_id, SMD_DEV_TYPE_DATA, &blob_id);
```

Comment:
Seems you need to rebase, the bio_blob_delete() in my 'vos_on_blob' branch accepts an extra 'device type' parameter.